src/ai/defaultai_seafaring.cc

```diff
@@ -254,24 +269,44 @@ bool DefaultAI::check_ships(const Time& gametime) {
 		// False indicates that we can postpone next call of this function
 		return false;
 	}
+	expeditions_in_progress = 0;
+	warships_count = 0;
+	tradeships_count = 0;
 
 	if (!allships.empty()) {
 		// iterating over ships and doing what is needed
 		for (ShipObserver& so : allships) {
 
+			if (so.ship->get_ship_type() == Widelands::ShipType::kWarship) {
+				++warships_count;
+				if (tradeship_refit_needed) {
+					game().send_player_refit_ship(*so.ship, Widelands::ShipType::kTransport);
+					tradeship_refit_needed = false;
+					--warships_count;
+				}
+			} else {
+				++tradeships_count;
+				if (warship_needed) {
```

Comment:
Not observed, but I suspect expedition ships may need protection:

```suggestion
				if (warship_needed && !so.ship->state_is_expedition()) {
```

Reply:
good idea thanks